extensions/amp-skimlinks/0.1/amp-skimlinks.js

```diff
@@ -0,0 +1,243 @@
+   * Make a fallback call to beacon if no links were founds on the page.
+   * Send the impression tracking once we have the beacon API response.
+   * @return {Promise}
+   * @private
+   */
+  onPageScanned_() {
+    // .firstRequest may be null if the page doesn't have any non-excluded
+    // links.
+    const beaconApiPromise = this.affiliateLinkResolver_.firstRequest ||
+        // If it's the case, fallback with manual call
+        this.affiliateLinkResolver_.fetchDomainResolverApi([]);
+
+    return beaconApiPromise.then(this.sendImpressionTracking_.bind(this));
+  }
+
+  /**
+   * Initialise Skimlinks LinkRewriter
+   * @return {!./link-rewriter/link-rewriter.LinkRewriter}
+   * @private
+   */
+  initSkimlinksLinkRewriter_() {
+    const options = {
+      linkSelector: this.skimOptions_.linkSelector,
+    };
+
+    const linkRewriter = this.linkRewriterService_.registerLinkRewriter(
+        SKIMLINKS_REWRITER_ID,
+        getBoundFunction(this.affiliateLinkResolver_, 'resolveUnknownAnchors'),
+        options
+    );
+
+    const eventHandlers = {
+      // We are only interested in the first page scan.
+      [linkRewriterEvents.PAGE_SCANNED]: once(this.onPageScanned_.bind(this)),
+      [linkRewriterEvents.CLICK]: this.onClick_.bind(this),
+    };
+
+    linkRewriter.events.add(event => {
+      const handler = eventHandlers[event.type];
+      if (handler) {
+        handler(event.eventData);
+      }
+    });
+
+    return linkRewriter;
+  }
+
+  /**
+   * Initialise tracking module
+   * @return {!./tracking.Tracking}
+   * @private
+   */
+  initTracking_() {
+    // 'amp-analytics' API is waiting for CommonSignals.LOAD_START to be
+    // triggered before sending requests.
+    // Normally CommonSignals.LOAD_START is sent from layoutCallback but since
+    // we are using layout = 'nodisplay', 'layoutCallback' is never called.
+    // We need to call it manually to have CustomEventReporterBuilder working.
+    this.signals().signal(CommonSignals.LOAD_START);
+    return new Tracking(
+        this.element,
+        this.skimOptions_
+    );
+  }
+
+  /**
+   * A click (left or right) on an anchor has happened,
+   * fire NA clicks if needed.
+   * @param {!{linkRewriterId: ?string, anchor: !HTMLElement, clickType: string}} eventData - click event information
+   * @private
+   */
+  onClick_(eventData) {
+    const doClickTracking = (
+      // Test two scenarios:
+      //  - Link hasn't been replaced at all: eventData.linkRewriterId === null
+      //  - Link has been replaced but not by Skimlinks:
+      //    E.g: eventData.linkRewriterId === "awin"
+      eventData.linkRewriterId !== SKIMLINKS_REWRITER_ID &&
+      // Also, context menu click should not send tracking
+      eventData.clickType !== 'contextmenu'
+    );
+
+    if (doClickTracking) {
+      this.trackingService_.sendNaClickTracking(eventData.anchor);
+    }
+  }
+
+  /** @override */
+  isLayoutSupported() {
+    return true;
+  }
+}
+
+
```

Comment:
nitpick. remove extra newline between methods here and throughout code. It used to be the style enforced but has changed.